code/game/objects/items/devices/window_painter.dm

```diff
@@ -0,0 +1,15 @@
+/obj/item/pipe_painter/window_painter
+	name = "window painter"
+	icon_state = "window_painter"
+
+/obj/item/pipe_painter/window_painter/afterattack(atom/A, mob/user as mob)
+	if(!(istype(A,/obj/structure/window/reinforced) || istype(A,/obj/structure/window/basic) || istype(A,/obj/machinery/door/window) || istype(A,/obj/structure/window/full/reinforced) || istype(A,/obj/structure/window/full/basic)) || !in_range(user, A))
```

Comment:
Having so many conditions like this in 1 if statement makes it a bit hard to read, and annoying to add to later. I suggest putting all these types in a list on the window painter, i.e
```dm
var/list/paintable_windows = list(
	/obj/structure/window/reinforced,
	/obj/structure/window/basic
	// etc.
)
```
And then you can do:
```dm
if(!is_type_in_list(A, paintable_windows) || !in_range(user, A))
	return
```